server/hedley/modules/custom/hedley_reports/hedley_reports.module

```diff
@@ -3416,6 +3407,969 @@ function hedley_reports_count_acute_illness_signs(array $values) {
+ *
+ * @param int $person_id
+ *   The ID of the person (child) whose scoreboard data is being
+ *   processed.
+ * @param DateTime $well_child_encounter_start_date_obj
+ *   The DateTime object representing the start date of the current
+ *   well-child encounter.
+ *
+ * @return array
+ *   A structured array of vaccination doses and dates, organized by
+ *   immunization type, based on child scoreboard encounters.
+ */
+function hedley_reports_well_child_generate_vaccination_progress_data_by_child_scoreboard($person_id, DateTime $well_child_encounter_start_date_obj) {
+  $return = [];
+  if (empty($person_id)) {
+    return $return;
+  }
+
+  $query = new EntityFieldQuery();
+  $query
+    ->entityCondition('entity_type', 'node')
+    ->entityCondition('bundle', 'individual_participant')
+    ->fieldCondition('field_encounter_type', 'value', 'child-scoreboard')
+    ->fieldCondition('field_person', 'target_id', $person_id)
+    ->propertyCondition('status', NODE_PUBLISHED)
+    ->addTag('exclude_deleted');
+
+  $result = $query
+    ->range(0, 1)
+    ->execute();
+
+  if (empty($result['node'])) {
+    // No more items left.
+    return $return;
+  }
+
+  $participant_id = key($result['node']);
+  // Load all encounters of current participant.
+  $query = new EntityFieldQuery();
+  $result = $query
+    ->entityCondition('entity_type', 'node')
+    ->entityCondition('bundle', 'child_scoreboard_encounter')
+    ->propertyCondition('status', NODE_PUBLISHED)
+    ->fieldCondition('field_individual_participant', 'target_id', $participant_id)
+    ->propertyOrderBy('nid')
+    ->execute();
+
+  if (empty($result['node'])) {
+    return $return;
+  }
+
+  $immunisation_types = [
+    'child_scoreboard_bcg_iz',
+    'child_scoreboard_dtp_iz',
+    'child_scoreboard_dtp_sa_iz',
+    'child_scoreboard_ipv_iz',
+    'child_scoreboard_mr_iz',
+    'child_scoreboard_opv_iz',
+    'child_scoreboard_pcv13_iz',
+    'child_scoreboard_rotarix_iz',
+  ];
+
+  $immunisations_by_type = [];
+  foreach ($immunisation_types as $immunisation_type) {
+    $immunisations_by_type[$immunisation_type] = [
+      'doses' => [],
+      'dates' => [],
+    ];
+  }
+  $encounters = node_load_multiple(array_keys($result['node']));
+  foreach ($encounters as $encounter) {
+    $start_date = explode(' ', $encounter->field_scheduled_date[LANGUAGE_NONE][0]['value'])[0];
+    $start_date_obj = new DateTime($start_date);
+    if ($start_date_obj > $well_child_encounter_start_date_obj) {
+      // This encounter started after well child encounter, so, skip it.
+      continue;
+    }
+
+    // Loading all immunisation activities recorded during the encounter.
+    $query = new EntityFieldQuery();
+    $result = $query
+      ->entityCondition('entity_type', 'node')
+      ->entityCondition('bundle', $immunisation_types, 'IN')
+      ->propertyCondition('status', NODE_PUBLISHED)
+      ->fieldCondition('field_child_scoreboard_encounter', 'target_id', $encounter->nid)
```

Comment:
Do we need the `exclude_deleted` tag here as well? If not, why do we need it for one of the above `EntityFieldQuery`s?

Reply:
Well, no, we don't need it here, as we query for measurements, and they don't have `field_deleted`.
This flag was introduces while we were developing merges of duplicate patients.
So, for example, patient was created twice, and each instances had one nutrition encounter.
So, when we merge them, we want to transfer all encounters and measurements from patient 1 to patient 2. The participant itself is not needed, as there should be single participant for patient in Nutrition (to which all encounters are pointing).
Therefore, participant should have `field_deleted` to mark patient 1 participant as deleted, but encounters and measurements should not, as they always get transferred.